recording_external_revenue/models/recording_external_revenue_raw.py

```diff
@@ -41,23 +39,282 @@ class RecordingExternalRevenueRaw(models.Model):
     product_external_catalog = fields.Char(index=True)
     product_external_catalog_reference = fields.Char(index=True)
 
-    quantity = fields.Float()
-    gross_amount_per_unit = fields.Float()
-    gross_amount = fields.Float()
-    revenue_id = fields.Many2one(
-        "recording.external.revenue",
-        readonly=True,
-    )
+    revenue_id = fields.Many2one("recording.external.revenue", readonly=True,)
     currency = fields.Char(index=True)
     tax = fields.Char()
     tax_base = fields.Selection(
         [
             ("net_amount", "Net Amount Before Tax"),
-            ("gross_amount", "Gross Amount Before Tax")
+            ("gross_amount", "Gross Amount Before Tax"),
         ],
     )
 
-    raw_unit_amount = fields.Float("Raw Unit Amount (Untaxed)")
-    raw_amount = fields.Float("Total Raw Amount (Untaxed)")
+    quantity = fields.Float()
+    gross_amount_per_unit = fields.Float()
+    gross_amount = fields.Float()
     commission_amount = fields.Float("Total Commissions Amount")
     net_amount = fields.Float("Total Net Amount (Untaxed)")
+
+    def make_revenue(self):
+        revenue = self.env["recording.external.revenue"].new()
+
+        for field in self.get_direct_mapping_fields():
+            revenue[field] = self[field]
+
+        revenue.partner_id = self._map_partner()
+        revenue.country_id = self._map_country()
+        revenue.state_id = self._map_country_state()
+        revenue.platform_id = self._map_platform()
+        revenue.subplatform_id = self._map_subplatform()
+        revenue.product_id = self._map_product()
+        revenue.recording_id = self._map_recording()
+        revenue.tax_id = self._map_tax()
+        revenue.currency_id = self._map_currency()
+        return revenue
+
+    def get_direct_mapping_fields(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-recording-addons/pullRequest?prid=5602996)